### PR TITLE
concurrency: add counters waiting for each other

### PR DESCRIPTION
**Pull request by davidkostov**

### CODE REVIEW

**Comment by mgenov** [comment]
Which task is this one ?

**Comment by mgenov** [comment]
Is this task 3: Да се модифицират нишките от предната задача така, че след всяко отброяване съответната нишка да спира и да позволява на другата да брои.
?

**Comment by davidkostov** [comment]
Yes.

File: src/main/java/com/clouway/threads3/Counter.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        this.name = name;
+    }
+
+    public void syncWithCounterThread(Thread threadToSyncWith) {
+        syncCounterThread = threadToSyncWith;
+    }
+
+    public void run() {
+        for (int i = from; i <= to; i++) {
+            synchronized (this) {
```
Comment (mgenov) [inline src/main/java/com/clouway/threads3/Counter.java]: Which is the common resource which is locked between threads ?

Comment (davidkostov) [reply src/main/java/com/clouway/threads3/Counter.java]: Should there be one? I thought they should both have their respective ranges, names and counts and the only interaction is when one finishes counting it terminates the other one. Correct me if i'm wrong please.

Comment (mgenov) [reply src/main/java/com/clouway/threads3/Counter.java]: Now you are doing it with timeouts this is why it's not required, but it enforces some time for doing of the job.

**Pull request merged**